pkg/cloud/amazon/s3_storage.go

```diff
@@ -657,7 +697,8 @@ func (u *putUploader) Write(p []byte) (int, error) {
 
 func (u *putUploader) Close() error {
 	u.input.Body = bytes.NewReader(u.b.Bytes())
-	_, err := u.client.PutObject(u.input)
+	// TODO(adityamaru): plumb a ctx through
```

Comment:
This is going to be a rather large change, what do you think about doing this in a follow up?